lib/web_ui/lib/src/engine/text/measurement.dart

```diff
@@ -958,3 +984,28 @@ class MaxIntrinsicCalculator {
     _lastHardLineEnd = hardLineEnd;
   }
 }
+
+double calculateAlignOffsetForLine({
+  @required EngineParagraph paragraph,
+  @required double lineWidth,
+  @required double maxWidth,
+}) {
+  final double emptySpace = maxWidth - lineWidth;
+  // WARNING: the [paragraph] may not be laid out yet at this point.
```

Comment:
Would be useful to know why it matters for this function? E.g. you can add "This function must not use layout metrics, such as paragraph.height"